rewrite-java-test/src/test/java/org/openrewrite/java/RecipeMarkupDemonstrationTest.java

```diff
@@ -40,4 +50,46 @@ class Test {
           )
         );
     }
+
+    @Test
+    void exceptionsWithoutCicularReferences() {
+        Recipe testRecipe = new TestRecipe();
+        rewriteRun(
+          spec -> spec.recipe(testRecipe),
+          java(
+            """
+              class Test {
+              }
+              """,
+            """
+              /*~~(this the parent of a circular exception)~~>*/class Test {
+              }
+              """)
+        );
+    }
+    @EqualsAndHashCode(callSuper = false)
+    static class TestRecipe extends Recipe {
+        @Override
+        public String getDisplayName() {
+            return "test recipe that produces an error";
+        }
+        @Override
+        public String getDescription() {
+            return "test recipe that produces an error using a circular exception cause.";
+        }
+        @Override
+        protected List<SourceFile> visit(List<SourceFile> before, ExecutionContext ctx) {
```

Comment:
For the OpenRewrite 8.0 upgrade it would be better if we here could implement a "normal" recipe implementing `getVisitor()` rather than `visit()`, as the latter requires modifications to compile. This is even easier by using the static `org.openrewrite.test.RewriteTest#toRecipe()` utility, as you then only need to supply the visitor rather than a full-blown recipe.